recipes/hidapi/all/conanfile.py

```diff
@@ -0,0 +1,126 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, MSBuild, tools
+import os, glob
+
+class HidapiConan(ConanFile):
+    name = "hidapi"
+    license = "BSD-Style"
+    homepage = "https://github.com/libusb/hidapi"
+    url = "https://github.com/conan-io/conan-center-index"
+    topics = ("conan", "hidapi", "usb", "device","hid")
+    description = "conan package for hidapi"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "minosx": ['10.7', '10.8', '10.9', '10.10', '10.11'],
+        "fPIC": [True, False],
+        "with_libusb": [True, False],
+        "shared": [True, False], 
+        "enable_udev": [True, False]
+    }
+    default_options = {
+        "minosx": 10.7, 
+        "fPIC": True, 
+        "with_libusb": False,
+        "shared": False,
+        "enable_udev": True
+    }
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return os.path.join(self.source_folder, "source_subfolder")        
+
+    @property
+    def _is_mingw(self):
+        return self.settings.os == "Windows" and self.settings.compiler == "gcc"
+
+    @property
+    def _is_msvc(self):
+        return self.settings.os == "Windows" and self.settings.compiler == "Visual Studio"
+
+    def config_options(self):
+        if self.settings.os != "Macos":
+            del self.options.minosx
+        if self.settings.os != "Linux":
+            del self.options.with_libusb
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def requirements(self):
+        if self.settings.os == "Linux" or self.settings.os == "FreeBSD":
+            self.requires("libusb/[~=1.0]")
```

Comment:
These two do no match

maybe allow FreeBSD above and use `self.options.get_safe("...")` ?